datafusion/expr/src/udaf.rs

```diff
@@ -195,6 +197,21 @@ impl AggregateUDF {
     pub fn create_groups_accumulator(&self) -> Result<Box<dyn GroupsAccumulator>> {
         self.inner.create_groups_accumulator()
     }
+    /// Do the function rewrite
+    ///
+    /// See [`AggregateUDFImpl::simplify`] for more details.
+    pub fn simplify(
+        &self,
+        args: Vec<Expr>,
```

Comment:
I think we should reuse args in AggregateFunction 
https://github.com/apache/datafusion/blob/8190cb97216e4f46faccbeddae57f6773587955f/datafusion/expr/src/expr.rs#L568-L582

Reply:
I don't know, mixed feelings about it. IMHO I see no benefits, would not simplify anything and another structure would be introduced.

Reply:
Do we change signature of `pub struct AggregateFunction` moving `args`, `distinct`, `filter` ... to `AggegateArgs` or we keep `AggegateArgs` as auxiliary structure?